### DISCUSSION TITLE
Service and Repository in Laravel MVC

### DISCUSSION THREAD

**Discussion by nicejjss** [laravel/framework#52498] [Q&A]
In some projects I found that Service will handle the logic of the app and return to the controller, the Repository will handle the logic and query of the model. What are they and when I need to use them?

**Comment by noefleury** [comment]
I think you are talking about the `Laravel Service-Repository Pattern`.

It's a good option if you want to maintain your logic clean and organized when interacting with database.

That's a nice structure !
